src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsIpc/IpcEndpointConfig.cs

```diff
@@ -0,0 +1,126 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Diagnostics.NETCore.Client
+{
+    internal class IpcEndpointConfig
+    {
+        public enum PortType
+        {
+            Connect,
+            Listen
+        }
+
+        PortType _type;
+
+        public string Address { get; }
+
+        public bool IsConnectConfig {
```

Comment:
Not really, but since we handle diagnosticPort as an empty string/null, being able to parse that and continue getting the address in places where we use it reduce the changes needed in the tooling code, so being amble to create an instance even when diagnosticPort was empty/null simplified that. I guess we could do more major changes to the whole process-id, name, diagnosticPort handling in the tooling, but maybe we should do that separately.

Reply:
> so being able to create an instance even when diagnosticPort was empty/null simplified that

OK, as long as it is useful I'm happy : ) Should the time come that we think this deserves to be part of the public API I'd step my scrutiny a good bit.

Reply:
Do you think it would be better to let the caller handle case of failure and then passing around a null ref through its methods to handle case where diagnostic port is empty/null? That change is simple to be done, making this class cleaner and we put the burden on the caller to better handle it? Would that pass your bar? :)

Reply:
> Do you think it would be better to let the caller handle case of failure and then passing around a null ref through its methods to handle case where diagnostic port is empty/null, like in DiagnosticsClientBuilder.Build?

I was speculating that it would be better but I hadn't gone to look at every usage and contemplate what the alternative code was going to look like. My assumption is that the callers already need to handle other exceptions that might be thrown from Parse() so presumably a FormatException saying address is empty has the same control flow needs as an exception saying the scheme is bad.

Reply:
Dropping the check for resume runtime actually simplified this process so we can move the parsing into  DiagnosticsClientBuilder.Build.

Reply:
@noahfalk, meeting the bar, https://github.com/dotnet/diagnostics/pull/2374/commits/f194079ce6820db9b43ecc56404381afb8d07857?